### PR TITLE
Feat/recent contracts

### PR DESCRIPTION
**Pull request by jennieramida**

### CODE REVIEW

File: src/lib/hooks/useContractFilter.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useCallback } from "react";
+
+import type { ContractInfo } from "lib/types";
+
+export const useContractSearchFilter = (keyword: string) =>
```
Comment (poomthiti) [inline src/lib/hooks/useContractFilter.ts:5]: I don't think this need to be a hook at all. We can delete this file move the logic to the same file as useRecentContractsData.

File: src/lib/pages/contracts/data.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    };
+  });
+
+  return useMemo(() => {
+    return {
+      recentContracts: recentContracts.filter(searchFilterFn),
+      isLoading,
+    };
+  }, [isLoading, recentContracts, searchFilterFn]);
+};
```
Comment (poomthiti) [inline src/lib/pages/contracts/data.ts:37]:
```tsx
import { useMemo } from "react";

import { useContractStore } from "lib/providers/store";
import { useContractListQuery } from "lib/services/contractService";
import type { ContractInfo } from "lib/types";

interface RecentContractsData {
  recentContracts: ContractInfo[];
  isLoading: boolean;
}

const searchFilterFn = (keyword: string) => (contract: ContractInfo) => {
  const computedKeyword = keyword.trim();
  if (!computedKeyword.length) return true;
  return (
    contract.contractAddress.startsWith(computedKeyword) ||
    contract.name?.toLowerCase().includes(computedKeyword.toLowerCase()) ||
    contract.label.toLowerCase().includes(computedKeyword.toLowerCase())
  );
};

export const useRecentContractsData = (
  keyword: string
): RecentContractsData => {
  const { getContractLocalInfo } = useContractStore();
  const { data: rawRecentContracts = [], isLoading } = useContractListQuery();

  const recentContracts = rawRecentContracts.map<ContractInfo>((contract) => {
    const contractLocalInfo = getContractLocalInfo(contract.contractAddress);
    return {
      ...contract,
      name: contractLocalInfo?.name,
      description: contractLocalInfo?.description,
      tags: contractLocalInfo?.tags,
      lists: contractLocalInfo?.lists,
    };
  });

  return useMemo(() => {
    return {
      recentContracts: recentContracts.filter(searchFilterFn(keyword)),
      isLoading,
    };
  }, [isLoading, recentContracts, keyword]);
};

```

File: src/lib/pages/contracts/index.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  const onRowSelect = (contract: ContractAddr) =>
+    navigate({
+      pathname: "/contracts/[contract]",
+      query: { contract },
+    });
+  const { watch, setValue } = useForm<RecentContractsState>({
+    defaultValues: {
+      keyword: "",
+    },
+  });
```
Comment (poomthiti) [inline src/lib/pages/contracts/index.tsx:30]: just use useState, useForm is not necessary here

File: src/lib/pages/contracts/index.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        <Heading
+          variant="h5"
+          as="h5"
+          minH="36px"
+          display="flex"
+          alignItems="center"
+        >
+          Recent Contracts
+        </Heading>
+        <Text variant="body2" color="text.dark" fontWeight="500">
```
Comment (poomthiti) [inline src/lib/pages/contracts/index.tsx:48]: {500}

File: src/lib/pages/contracts/index.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            message={
+              isSearching
+                ? "No matched contracts found"
+                : "Most recent 100 contracts will display here."
+            }
+            withBorder
+          />
+        }
+        onRowSelect={onRowSelect}
+        hasTag={false}
```
Comment (poomthiti) [inline src/lib/pages/contracts/index.tsx:77]: just `withoutTag`

File: src/lib/pages/my-codes/index.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      query: { codeId },
+    });
+
+  const { watch, setValue } = useForm<CodeFilterState>({
+    defaultValues: {
+      permissionValue: "all",
+      keyword: "",
+    },
+  });
+  const { keyword, permissionValue } = watch();
```
Comment (poomthiti) [inline src/lib/pages/my-codes/index.tsx:47]: prefer to have this refactored to useState, but just leaving a TODO is fine

**Review by poomthiti — CHANGES_REQUESTED** [review]

File: src/lib/pages/contracts/index.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  const onRowSelect = (contract: ContractAddr) =>
+    navigate({
+      pathname: "/contracts/[contract]",
+      query: { contract },
+    });
+  const [searchKeyword, setSearchKeyword] = useState("");
+
+  const { recentContracts, isLoading } = useRecentContractsData(searchKeyword);
+
+  const isSearching = !!searchKeyword;
```
Comment (bkioshn) [inline src/lib/pages/contracts/index.tsx:25]: ditto

File: src/lib/components/table/contracts/ContractsTable.tsx [hunk 8]
```diff
@@ -22,20 +23,28 @@ export const ContractsTable = ({
   onRowSelect,
   isReadOnly = false,
   withCTA,
+  withoutTag = false,
```
Comment (poomthiti) [inline src/lib/components/table/contracts/ContractsTable.tsx:26]: remove default

File: src/lib/components/table/contracts/ContractsTableHeader.tsx [hunk 9]
```diff
@@ -9,15 +9,17 @@ export const ContractsTableHeader = ({
   templateColumns,
   isReadOnly,
   withCTA,
+  withoutTag = false,
```
Comment (poomthiti) [inline src/lib/components/table/contracts/ContractsTableHeader.tsx:12]: remove default

File: src/lib/components/table/contracts/ContractsTableRow.tsx [hunk 10]
```diff
@@ -92,6 +93,7 @@ export const ContractsTableRow = ({
   onRowSelect,
   isReadOnly,
   withCTA,
+  withoutTag = false,
```
Comment (poomthiti) [inline src/lib/components/table/contracts/ContractsTableRow.tsx:96]: ditto

**Review by poomthiti — CHANGES_REQUESTED** [review]

**Review by poomthiti — APPROVED** [review]

File: src/lib/pages/contract-details/components/ContractTop.tsx [hunk 11]
```diff
@@ -97,19 +98,21 @@ export const ContractTop = ({ contractData }: ContractTopProps) => {
...
-        )}
+        <Breadcrumb
+          items={[
+            {
+              text: publicName ? "Public Projects" : "Contracts",
+              href: publicName ? "/projects" : "/contracts",
+            },
+            {
+              text: publicName,
+              href: `/projects/${publicProject.publicInfo?.slug}`,
```
Comment (evilpeach) [inline src/lib/pages/contract-details/components/ContractTop.tsx:109]: What happens if publicName is undefined.

File: src/lib/pages/my-codes/index.tsx [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+
+import { MySavedCodesSection } from "./components/MySavedCodesSection";
+import { MyStoredCodesSection } from "./components/MyStoredCodesSection";
+
+interface CodeFilterState {
+  keyword: string;
+  permissionValue: PermissionFilterValue;
+}
+
+const Codes = observer(() => {
```
Comment (evilpeach) [inline src/lib/pages/my-codes/index.tsx:32]: Rename to `MyCodes`

File: src/lib/pages/my-codes/index.tsx [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+              isSearching={isSearching}
+            />
+          </TabPanel>
+        </TabPanels>
+      </Tabs>
+    </PageContainer>
+  );
+});
+
+export default Codes;
```
Comment (evilpeach) [inline src/lib/pages/my-codes/index.tsx:138]: rename it as well

**Review by evilpeach — APPROVED** [review]

**Pull request merged**